substrate/frame/asset-conversion/src/tests.rs

```diff
@@ -64,40 +66,71 @@ fn pool_assets() -> Vec<u32> {
 	s
 }
 
-fn create_tokens(owner: u128, tokens: Vec<NativeOrAssetId<u32>>) {
+fn create_tokens(owner: u128, tokens: Vec<NativeOrWithId<u32>>) {
 	create_tokens_with_ed(owner, tokens, 1)
 }
 
-fn create_tokens_with_ed(owner: u128, tokens: Vec<NativeOrAssetId<u32>>, ed: u128) {
+fn create_tokens_with_ed(owner: u128, tokens: Vec<NativeOrWithId<u32>>, ed: u128) {
 	for token_id in tokens {
-		let MultiAssetIdConversionResult::Converted(asset_id) =
-			NativeOrAssetIdConverter::try_convert(&token_id)
-		else {
-			unreachable!("invalid token")
+		let asset_id = match token_id {
+			NativeOrWithId::WithId(id) => id,
+			_ => unreachable!("invalid token"),
 		};
 		assert_ok!(Assets::force_create(RuntimeOrigin::root(), asset_id, owner, false, ed));
 	}
 }
 
-fn balance(owner: u128, token_id: NativeOrAssetId<u32>) -> u128 {
-	match token_id {
-		NativeOrAssetId::Native => <<Test as Config>::Currency>::free_balance(owner),
-		NativeOrAssetId::Asset(token_id) => <<Test as Config>::Assets>::balance(token_id, owner),
-	}
+fn balance(owner: u128, token_id: NativeOrWithId<u32>) -> u128 {
+	<<Test as Config>::Assets>::balance(token_id, &owner)
 }
 
 fn pool_balance(owner: u128, token_id: u32) -> u128 {
 	<<Test as Config>::PoolAssets>::balance(token_id, owner)
 }
 
-fn get_ed() -> u128 {
-	<<Test as Config>::Currency>::minimum_balance()
+fn get_native_ed() -> u128 {
+	<<Test as Config>::Assets>::minimum_balance(NativeOrWithId::Native)
 }
 
 macro_rules! bvec {
-	($( $x:ident ),*) => {
+	($($x:expr),+ $(,)?) => (
 		vec![$( Box::new( $x ), )*]
-	}
+	)
+}
+
+#[test]
+fn validate_with_first_asset_pool_id_locator() {
+	new_test_ext().execute_with(|| {
+		use NativeOrWithId::{Native, WithId};
+		assert_eq!(WithFirstAssetLocator::pool_id(&Native, &WithId(2)), Ok((Native, WithId(2))));
```

Comment:
`WithFirstAssetLocator` - shouldn't it be 'WithFirst**Native**Locator'?

Reply:
ok, I see where it comes from. I would put a type alias like `type NativeFirstLocator = WithFirstAssetLocator<...>`